extensions/kubeconfig/podlogs.go

```diff
@@ -12,8 +14,14 @@ import (
 	restclient "k8s.io/client-go/rest"
 )
 
-// GetPodLogs is a helper function that resturns logs from a pod using rest client
-func GetPodLogs(client *rancher.Client, clusterID string, podName string, namespace string) (string, error) {
+// GetPodLogs is a function that fetches the logs of a specific pod within a
+// Kubernetes cluster. It requires a rancher client instance, clusterID,
+// podName, namespace, and a buffer size represented as a string. The buffer
+// size should include a suffix of 'KB', 'MB', or 'GB'. If a blank string is
+// provided, no buffering will be used when reading log lines. The function
+// returns a string which contains all the logs and an error if any occurred
+// during the operation.
+func GetPodLogs(client *rancher.Client, clusterID string, podName string, namespace string, bufferSizeStr string) (string, error) {
```

Comment:
why do we want bufferSizeStr here?

Reply:
Will create a PR into rancher/rancher once shepherd is merged. Since rancher/rancher is still using internal extensions. Thanks for the reminder! 🚀